File: library/src/com/sromku/simple/storage/helpers/SizeUnit.java
package com.sromku.simple.storage.helpers;

public enum SizeUnit {
	B(1),
	KB(SizeUnit.BYTES),
	MB(SizeUnit.BYTES * SizeUnit.BYTES),
	GB(SizeUnit.BYTES * SizeUnit.BYTES * SizeUnit.BYTES),
	TB(SizeUnit.BYTES * SizeUnit.BYTES * SizeUnit.BYTES * SizeUnit.BYTES);

	private long inBytes;
	private static final int BYTES = 1024;

	private SizeUnit(long bytes) {
		this.inBytes = bytes;
	}

	public long inBytes() {
		return inBytes;
	}
}

File: storage/src/main/kotlin/com/snatik/storage/SizeUnit.kt
package com.snatik.storage

import java.util.Locale

/** Binary size units, 1024-based. */
public enum class SizeUnit(public val bytes: Long) {
    B(1L),
    KB(1L shl 10),
    MB(1L shl 20),
    GB(1L shl 30),
    TB(1L shl 40);

    /** Express [byteCount] in this unit. */
    public fun convert(byteCount: Long): Double = byteCount.toDouble() / bytes

    public companion object {
        /**
         * Human readable size such as `512 B`, `1.5 KB`, `20.3 MB`.
         */
        public fun readable(byteCount: Long): String {
            val unit = entries.lastOrNull { byteCount >= it.bytes } ?: B
            return if (unit == B) {
                "$byteCount B"
            } else {
                String.format(Locale.US, "%.1f %s", unit.convert(byteCount), unit.name)
            }
        }
    }
}

/** Human readable size, see [SizeUnit.readable]. */
public fun Long.toReadableSize(): String = SizeUnit.readable(this)
